controller/space_codebases.go

```diff
@@ -58,6 +60,9 @@ func (c *SpaceCodebasesController) Create(ctx *app.CreateSpaceCodebasesContext)
 		}
 		err = appl.Codebases().Create(ctx, &newCodeBase)
 		if err != nil {
+			if gormsupport.IsUniqueViolation(err, "codebases_spaceid_url_idx") {
+				return errors.NewBadParameterError("Codebase URL", *reqIter.Attributes.URL).Expected("unique")
+			}
```

Comment:
The error being returned by `appl.Codebases().Create(ctx, &newCodeBase)` is not being parsed here, any help would be appreciated.

Reply:
@surajssd just debug where it fails. It could be a different error code for example.

Reply:
@surajssd , @pranavgore09   is right. You shouldn't handle such an error on controller level. All errors from repository are wrapped into our own `errors.XY` errors and the postgres error is hidden underneath.

Reply:
thanks @kwk @pranavgore09 for all the help :)